### PR TITLE
✨(forum) enables latex mathematical syntax

### PR DESCRIPTION
**Pull request by carofun**
## Purpose

Using katex, we enable latex mathematical syntax in draft.js to write equations in forums.
In draft.js press $ for starting Inline TeX and to write the $ character, type \$

## Proposal

https://user-images.githubusercontent.com/76939263/208128684-8e75b275-d538-4fce-9a9c-935cf81c56a4.mov

- disable code-editor plugin
- enabling draft-js-latex plugin

### CODE REVIEW

**Review by jbpenrath — COMMENTED** [review]
What do you think to add a button within the toolbar to add formula ?

File: src/frontend/js/utils/latex.ts [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import katex from 'katex';
 
 export const renderLatex = () => {
-  document.querySelectorAll('span.latex').forEach((math) => {
+  document.querySelectorAll('[class^=latex]').forEach((math) => {
```
[the remark this answers is not in the document]
Comment (carofun) [reply src/frontend/js/utils/latex.ts:4]: wdyt of ashley-custom-latex instead?

**Comment by carofun** [comment]
yes i think it's a good idea, maybe I can add it in another PR

File: src/frontend/js/utils/latex.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jbpenrath) [reply src/frontend/js/utils/latex.ts]: That's fine for me. Even `ashley-latex`

File: src/ashley/editor/decorators.py [hunk 2]
```diff
@@ -93,3 +93,28 @@ def image(props):
...
+    """
+    return DOM.create_element(
+        "span", {"class": "ashley-custom-latex-inline"}, props.get("tex", None)
+    )
+
+
+def render_children(props):
+    """
+    Decorator for the blocks in Draft.js ContentState. TEXBLOCK is a custom
+    block that is used through atomic blocs.
```
Comment (jbpenrath) [inline src/ashley/editor/decorators.py:110]:
```suggestion
    block that is used through atomic blocks.
```

File: src/frontend/jest.config.js [hunk 3]
```diff
@@ -4,5 +4,7 @@ module.exports = {
   setupFilesAfterEnv: ['./testSetup.ts', 'regenerator-runtime/runtime'],
   testMatch: [__dirname + '/js/**/*.spec.+(ts|tsx|js)'],
   testURL: 'https://localhost',
-  transformIgnorePatterns: ['node_modules/(?!(lodash-es)/)'],
+  transformIgnorePatterns: [
+    '/node_modules/(?!(' + 'lodash-es|' + 'draft-js-latex-plugin' + ')/)',
```
Comment (jbpenrath) [inline src/frontend/jest.config.js:8]: https://github.com/openfun/richie/blob/9bed95e7320d1d8c2e7f123c7b638f4a6b612f91/src/frontend/jest.config.js#L17

This syntax should work.
```suggestion
    '/node_modules/(?!(lodash-es|draft-js-latex-plugin)/)',
```

**Review by jbpenrath — APPROVED** [review]
Some nitpicks then LGTM ✅ !

**Pull request merged**